osfoffline/views/start_screen.py

```diff
@@ -21,47 +32,63 @@ def __init__(self):
+        osf = OSFQuery(loop, personal_access_token)
+        logging.debug('url: {}'.format(url))
         try:
-            user = session.query(User).filter(User.osf_login == user_name).one()
-            user.logged_in = True
-            save(session, user)
-            self.close()
-        except MultipleResultsFound:
-            logging.warning(
-                'multiple users with same username. deleting all users with this username. restarting function.')
-            for user in session.query(User):
-                if user.osf_login == user_name:
+            resp = yield from osf.make_request(url, expects=[200])
+        except (aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError, aiohttp.errors.TimeoutError):
+            # No internet connection
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Unable to connect to server. Check your internet connection or try again later."
+            )
+        except aiohttp.errors.HttpBadRequest:
+            # Invalid credentials
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Invalid login credentials."
+            )
+        else:
+            json_resp = yield from resp.json()
+            remote_user = RemoteUser(json_resp['data'])
+
+            try:
+                user = session.query(User).filter(User.osf_id == remote_user.id).one()
+            except MultipleResultsFound:
+                logging.warning('multiple users with same username. deleting all users with this username. restarting function.')
+                for user in session.query(User).filter(User.osf_id == remote_user.id).all():
                     session.delete(user)
                     save(session)
-            self.log_in()
-        except NoResultFound:
-            logging.warning('user doesnt exist. Creating user. and logging them in.')
-            user = User(
-                full_name=full_name,
-                osf_id=osf_id,
-                osf_login=user_name,
-                osf_local_folder_path='',
-                oauth_token=oauth_token,
-                osf_password=password
-            )
+                self.log_in()
+            except NoResultFound:
+                logging.debug('user doesnt exist. Creating user. and logging them in.')
+                user = User(
+                    full_name=remote_user.full_name,
+                    osf_id=remote_user.id,
+                    osf_login='',  # TODO: email goes here when more auth methods are added, not currently returned by APIv2
+                    osf_local_folder_path='',
+                    oauth_token=personal_access_token,
+                )
+            else:
+                if not user.oauth_token == personal_access_token:
+                    user.oauth_token == personal_access_token
+
             user.logged_in = True
             save(session, user)
```

Comment:
The possibility of a database save leaving an exception is the last thing to be addressed. It may have been missed because the conversation built on an outdated diff; see email notification for the UI suggestion.